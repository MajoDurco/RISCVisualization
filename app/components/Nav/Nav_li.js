import styled from 'styled-components';

export const Nav_li = styled.li`
	& a {
		text-align: center;
		padding: 14px 16px;
		text-decoration: none;
		color: white;
	};

	&:hover {
		background-color: #111;
	}
`
