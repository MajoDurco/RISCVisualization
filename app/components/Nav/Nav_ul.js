import styled from 'styled-components';

const Nav_ul = styled.ul`
	list-style-type: none;
	margin: 0;
	padding: 0;
	overflow: hidden;
	margin-left: 15px;
`

export default Nav_ul;
