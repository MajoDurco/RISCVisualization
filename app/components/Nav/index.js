import React from 'react'
import styled from 'styled-components'

import cpu from './cpu.png'

const Navig = styled.div`
	margin: 0;
	padding: 0;
	height: 40px;
	width: 100%;

	& * {
		display: inline;
	}
`

const ImgCpu = styled.img`
	height: inherit;
	width: 50px;
	margin-right: 15px;
`

function Nav() {
  return (
		<Navig>
			<ImgCpu src={cpu} />
			<h3>RISC Visualization Tool</h3>
		</Navig>
  );
}

export default Nav
