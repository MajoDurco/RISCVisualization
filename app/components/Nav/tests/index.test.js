import React from 'react'
import renderer from 'react-test-renderer'

import Nav from '../index'

it('nav renders correctly', () => {
  const navigation_bar = renderer.create(
    <Nav />
  ).toJSON();
  expect(navigation_bar).toMatchSnapshot();
});
