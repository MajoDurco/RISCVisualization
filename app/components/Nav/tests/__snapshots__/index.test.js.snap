exports[`test nav renders correctly 1`] = `
<div
  className="dVZZbp">
  <a
    className="fgURVR"
    onClick={[Function]}
    style={Object {}}>
    <img
      className="jaiOvz"
      src="IMAGE_MOCK" />
    <h1
      className="jbaQLL"
      size={undefined}>
      RISC Visualization Tool
    </h1>
  </a>
</div>
`;
